### PR TITLE
Kereru Testing

### PR DESCRIPTION
**Pull request by UberMouse**
I have

- Improved all of the existing RSpecs where needed
- Written the few new ones that needed to be (Record model and two controllers)
- Configured Rubocop (does not work on Travis yet)
- Installed YARD
- Written feature specs for registrations and logins

The only things left to be done are finishing off the feature specs, remaining items
- search interactions
- filter interactions
- user set interactions
- record show page interactions

[Review on Reviewable](https://reviewable.io/reviews/digitalnz/supplejack_website/15)

### CODE REVIEW

File: app/controllers/user_sets_controller.rb [hunk 1]
```diff
@@ -22,20 +21,16 @@ def index
...
-        render json: {sets: sets}
-      end
-    end
+    render json: {sets: sets}
   end
 
   def create
     if params[:user_set].nil?
-      render nothing: true, status: 500
+      return render nothing: true, status: 500
```
Comment (Reinkaos) [inline app/controllers/user_sets_controller.rb]: Could be a one line 'return render nothing: true, status: 500 if params[:user_set].nil?'

File: spec/controllers/user_sets_controller_spec.rb [hunk 2]
```diff
@@ -8,31 +8,76 @@
...
+
+    it "retrieves the sets from the current_sj_user" do
+      dummy_sj_user = instance_double(Supplejack::User)
+      
+      expect(controller).to receive(:current_sj_user).and_return(dummy_sj_user)
+      expect(dummy_sj_user).to receive(:sets).and_return(
+        [double(name: 'test', id: 1, count: 1)]
+      )
+
+      get :index
```
Comment (Reinkaos) [inline spec/controllers/user_sets_controller_spec.rb]: Why is the get after the expect?

File: spec/models/search_spec.rb [hunk 3]
```diff
@@ -7,6 +7,4 @@
 # http://digitalnz.org/supplejack
 
 RSpec.describe Search do
```
Comment (Reinkaos) [inline spec/models/search_spec.rb]: Has this been removed completely?

File: spec/controllers/user_sets_controller_spec.rb (hunk 2 above)
Comment (UberMouse) [inline spec/controllers/user_sets_controller_spec.rb]: The expects set up expectations in how the objects will be called, the get then triggers the calls. Which is the opposite of a normal test.

**Pull request merged**